src/main/java/dev/codesupport/web/common/security/JwtRequestFilter.java

```diff
@@ -39,71 +46,108 @@ public JwtRequestFilter(AuthenticationUserDetailsService userDetailsService) {
-                }
+    void checkForJWToken(HttpServletRequest request) {
+        String tokenString = getTokenFromRequest(request);
+        // If token is empty, skip all this, could be an authentication post request.
+        if (!StringUtils.isEmpty(tokenString)) {
+            validateJWToken(tokenString, request);
+        }
+    }
+
+    /**
+     * Validate token and configure Spring Security Context
+     *
+     * @param tokenString The token to validate
+     * @param request     The associated request
+     */
+    void validateJWToken(String tokenString, HttpServletRequest request) {
+        Assert.notNull(tokenString, "tokenString can not be null");
+        try {
+            // Get token from request, throws InvalidTokenException if invalid
+            JsonWebToken jwToken = jsonWebTokenFactory.createToken(tokenString);
+
+            String username = jwToken.getUsername();
+
+            configureSpringSecurityContext(username, request);
+        } catch (InvalidTokenException e) {
+            // Doesn't matter, just log it for debugging and the service with return an unauthorized error.
+            if (log.isDebugEnabled()) {
+                log.debug("Failed to validate token", e);
             }
         }
     }
 
+    /**
+     * Configures the Spring Security Context
+     * <p>If not already configured, this method configures the Spring Security Context, which is
+     * used later by the framework for access checks, etc.</p>
+     *
+     * @param username The user to be configured in the security context
+     * @param request  The requested associated to the configuration
+     */
+    void configureSpringSecurityContext(String username, HttpServletRequest request) {
+        Assert.notNull(username, "username can not be null");
+
+        // If authentication is not already configured
+        if (SecurityContextHolder.getContext().getAuthentication() == null) {
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+
+            UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities());
+            usernamePasswordAuthenticationToken
+                    .setDetails(new WebAuthenticationDetailsSource().buildDetails(request));
+
+            SecurityContextHolder.getContext().setAuthentication(usernamePasswordAuthenticationToken);
+        }
+    }
+
     /**
      * Extract token from header.
      *
      * @param request Corresponding request to extract token from
      * @return The raw token ascii string
      */
     String getTokenFromRequest(HttpServletRequest request) {
-        final String requestTokenHeader = request.getHeader("Authorization");
+        final String requestAuthorizationHeader = request.getHeader("Authorization");
         String jwtToken;
```

Comment:
Why not just make the default value `null`?

Reply:
It can be, it wastes a process cycle or two to do so, so it would just be slightly slower, though you probably wouldn't notice.  Generally you try not to initialize values twice though.

Reply:
I thought Java default initializes values regardless? Aren't you already losing that cycle anyway, and then re-assigning to `null` later?

Reply:
Just saw your Discord message regarding this, interesting, I always thought Java always default initialized. Resolving this.